tpm2_eventlog: Format partitions correctly

josephlr opened the pull request:
Fixes #2746 

Now each partition is printed separately, and we correctly display the UTF16 strings in the partition names.

This change required creating a helper struct:
```c
typedef struct {
    char16_t c;
} PACKED UTF16_CHAR;
```
so that we could pass the address of our UFT16 buffer to a helper function without invoking undefined behavior.

I tested this on some sample eventlogs and it now seems to produce correct output (with `--eventlog-version=2`).

Signed-off-by: Joe Richey <joerichey@google.com>

idesai commented:
@josephlr thank you. Can you also please add the eventlog binary you tested on to the [test-fixtures](https://github.com/tpm2-software/tpm2-tools/tree/master/test/integration/fixtures) and update the integration tests?

idesai commented on lib/efi_event.h:
```diff
@@ -147,7 +152,7 @@ typedef struct {
     UINT64 EndingLBA;
     UINT64 Attributes;
-    BYTE PartitionName[72];
+    UTF16_CHAR PartitionName[36];
```
@josephlr, what is the magic number 36?

josephlr replied:
That's just the definition in the EFI spec:
```c
typedef struct {
    EFI_GUID PartitionTypeGUID;
    EFI_GUID UniquePartitionGUID;
    EFI_LBA StartingLBA;
    EFI_LBA EndingLBA;
    UINT64 Attributes;
    CHAR16 PartitionName[36];
} PACKED EFI_PARTITION_ENTRY;
```

See section `5.3.3` of the [UEFI Spec v2.9](https://uefi.org/sites/default/files/resources/UEFI_Spec_2_9_2021_03_18.pdf).

The GPT format chose `36` so that `sizeof(EFI_PARTITION_ENTRY) == 128` ([godbolt link](https://godbolt.org/z/4hxbsYja6)) so that the entry fits exactly in two x86 cache lines.

josephlr replied:
@idesai, do you want me to add a comment explaining this, or is that implied from the structure of this file?

idesai replied:
@josephlr yes please.

josephlr replied:
Done

josephlr commented:
> @josephlr thank you. Can you also please add the eventlog binary you tested on to the [test-fixtures](https://github.com/tpm2-software/tpm2-tools/tree/master/test/integration/fixtures) and update the integration tests?

Added some additional Event Logs, and updated the integration tests to use `--eventlog-version=2` for those logs. I also updated the `.gitignore` and `test/.gitignore` files to properly include/exclude generated files (my change #2743 was not correct in this regard).

josephlr commented:
@idesai it looks like the `master-tss-build-test`s are failing for unrelated reasons.

idesai commented:
> @idesai it looks like the `master-tss-build-test`s are failing for unrelated reasons.

@josephlr, the CI is now fixed. Can you please rebase your PR on current master.
Thank you for adding the test fixtures and updating the test.

josephlr commented:
> @josephlr, the CI is now fixed. Can you please rebase your PR on current master.

Done, thanks for fixing the CI, looks like everything is passing now.
